extensions/interactions/TextInput/directives/text-input-validation.service.ts

```diff
@@ -111,15 +111,29 @@ export class TextInputValidationService {
     let seenStringsStartsWith: string[] = [];
     let seenStringsEquals: string[] = [];
     let seenStringsFuzzyEquals: string[] = [];
+    let seenRuleTypes: Set<string>;
 
     for (let [answerGroupIndex, answerGroup] of answerGroups.entries()) {
+      seenRuleTypes = new Set();
       for (let [ruleIndex, rule] of answerGroup.rules.entries()) {
-        let currentString = <string> rule.inputs.x;
+        if (seenRuleTypes.has(rule.type)) {
+          warningsList.push({
+            type: AppConstants.WARNING_TYPES.ERROR,
+            message: (
+              `Answer group ${answerGroupIndex + 1} has multiple rules on ` +
+              `the same type (${rule.type}) within the same group.`
+            )
+          });
+        }
+        seenRuleTypes.add(rule.type);
+
+
+        let currentStrings = <string[]> rule.inputs.x;
         if (rule.type === 'Contains') {
           // Check if the current string contains any of the previously seen
           // strings as a substring.
           if (seenStringsContains.some(
-            (seenString) => currentString.includes(seenString)) ||
+            (seenString) => currentStrings.includes(seenString)) ||
```

Comment:
This seems wrong? Previously, the variable was a string, now it's an array. I think you're checking for something different here?